### PR TITLE
Disallow root disk to be used for cache drives

### PR DESCRIPTION
**Pull request by poornas**
## How to test this PR?
Set MINIO_CACHE_DRIVES - should fail if root disk specified.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

File: internal/config/cache/config.go [hunk 1]
```diff
@@ -115,11 +118,20 @@ func parseCacheDrives(drives string) ([]string, error) {
 			endpoints = append(endpoints, d)
 		}
 	}
-
+	isCICD := os.Getenv("MINIO_CI_CD") != "" || os.Getenv("CI") != ""
 	for _, d := range endpoints {
 		if !filepath.IsAbs(d) {
 			return nil, config.ErrInvalidCacheDrivesValue(nil).Msg("cache dir should be absolute path: %s", d)
 		}
+		if runtime.GOOS != "windows" && !isCICD {
```
Comment (harshavardhana) [inline internal/config/cache/config.go]: IsRootDisk() handles windows so its  okay to only have this check when CICD is false.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Review by vadmeste — APPROVED** [review]
LGTM

**Comment by klauspost** [comment]
@harshavardhana @poornas 

It seems user hostile to disable it on upgrade.

Could we add a strongly worded warning and disable it in 6 months?

**Comment by harshavardhana** [comment]
> It seems user hostile to disable it on upgrade.

We can @klauspost  - we can support existing settings, and warn about it. however, disallow setting newer values.

**Comment by klauspost** [comment]
@harshavardhana Yeah, it may require hardware changes to fix. It is not a critical error that will (short term) cause data corruption, so it shouldn't kill the server on upgrade, but we shouldn't allow it for new/changed installs.

**Pull request merged**